src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -116,8 +119,9 @@ public static void after() throws Exception {
 
     /**
      * Subclass should handle assertions on the lucene query produced by the query builder under test here
+     * @param context
      */
-    protected abstract void assertLuceneQuery(QB queryBuilder, Query query) throws IOException;
+    protected abstract void assertLuceneQuery(QB queryBuilder, Query query, QueryParseContext context) throws IOException;
```

Comment:
the more I see QueryParseContext used in toQuery the more I think we should start splitting it in two objects, one for parsing and one for toQuery only. But maybe I am jumping ahead a bit, we can in the meantime familiarize with what it is used for and once we migrated all the queries go and do that other step? What do you think?